src/gravel/controllers/nodes/mgr.py

```diff
@@ -435,11 +435,9 @@ def _bootstrap_process():
             asyncio.set_event_loop(loop)
             loop.run_until_complete(_run_etcd())
 
-        process = multiprocessing.Process(
-            target=_bootstrap_process
-        )
-        process.start()
-        logger.info(f"started etcd process pid = {process.pid}")
+        t = threading.Thread(target=_bootstrap_process)
+        t.start()
+        logger.info("started etcd thread")
```

Comment:
"Starting it as a process, for some reason, allows etcd to squash the event loop's signal handlers, which breaks our "on shutdown" cleanup." Huh? As a separate process it shouldn't be able to do anything to the parent, should it? And if it can mess up the parent _process_, why wouldn't it also mess up the process when run as a thread? (I kinda feel like etcd should really be a separate process, just because it's its own daemon, rather than being some class or library that we import...)

Reply:
From the docs, that seems to be the daemon's instance name. `initial-cluster` seems much like `mon_initial_cluster` from Ceph. Even if that might set the cluster's name to the name of the first node, maybe there's a way to do that without having to name the first node something that does not match the hostname? (I'll look into that)

Reply:
wtf - this last comment should have been a reply to something else entirely. github seems to be on fire today.

Anyhoo, to @tserong actual comment:



> "Starting it as a process, for some reason, allows etcd to squash the event loop's signal handlers, which breaks our "on shutdown" cleanup." Huh? As a separate process it shouldn't be able to do anything to the parent, should it? And if it can mess up the parent _process_, why wouldn't it also mess up the process when run as a thread? (I kinda feel like etcd should really be a separate process, just because it's its own daemon, rather than being some class or library that we import...)

Yeah, I know. This is weird, really. I assumed the same thing, but for some reason, when starting an asyncio subprocess from a multithreading.Process, the signal handler is squashed; when starting the asyncio subprocess from a threading.Thread, it does not. I'll try getting a script to reproduce the behavior just to ensure I'm not imagining things though.

Reply:
Okay, I'm struggling to reproduce this with a script because it works fine with a Process. I truly can't tell wtf was going on. I'll try reverting it and see what happens, I guess.